### PR TITLE
config: Rename Configuration trait to OnlineConfig

### PR DESCRIPTION
**Pull request by andylokandy**
### What problem does this PR solve?

The name of `Configuration` trait and its deriving macro is very mysterious.

### What is changed and how it works?

Rename it to `OnlineConfig`

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2017 TiKV Project Authors. Licensed under Apache-2.0.
 
-//! Configuration for the entire server.
+//! OnlineConfig for the entire server.
```
Comment (NingLin-P) [inline src/config.rs:3]: This  line should not be changed

**Review by NingLin-P — COMMENTED** [review]
Rest LGTM, thanks!

**Review by zhongzc — APPROVED** [review]

**Comment by andylokandy** [comment]
@NingLin-P PTAL

**Review by hicqu — APPROVED** [review]

**Comment by hi-rustin** [comment]
@ti-chi-bot

**Pull request merged**